xml/smb_native_os.xml

```diff
@@ -351,4 +351,196 @@
     <param pos="0" name="service.vendor" value="Wind River"/>
     <param pos="0" name="service.product" value="VxWorks CIFS"/>
   </fingerprint>
+
+  <fingerprint pattern="^OS/400 \D(\d+)\D(\d+)\D(\d+)">
```

Comment:
For this and most others, it might be worth adding the trailing `$` to the regex.